### PR TITLE
Lazy connect when booting app

### PR DESCRIPTION
**Pull request by jannyHou**
Connect to strongloop/loopback#2242

When booting app from swagger generator, datasources are setup with ds.lazyConnect = true to avoid connection.

### CODE REVIEW

**Comment by raymondfeng** [comment]
Can we add a test case to verify `lazyConnect` works?

File: lib/mssql.js [hunk 1]
```diff
@@ -16,12 +16,16 @@ exports.initialize = function initializeSchema(dataSource, callback) {
   debug('Settings: %j', settings);
   var driver = new MsSQL(settings);
   dataSource.connector = driver;
-  driver.connect(function(err, connection) {
-    dataSource.client = connection;
-    dataSource.connector.dataSource = dataSource;
-    dataSource.connector.tableNameID = dataSource.settings.tableNameID;
-    callback && callback(err, connection);
-  });
+  if (settings.lazyConnect) {
```
Comment (raymondfeng) [inline lib/mssql.js]: We can execute the following before checking `settings.lazyConnect`:
```
dataSource.connector = driver;
dataSource.connector.dataSource = dataSource;
dataSource.connector.tableNameID = dataSource.settings.tableNameID;
```

File: lib/mssql.js [hunk 2]
```diff
@@ -16,12 +16,16 @@ exports.initialize = function initializeSchema(dataSource, callback) {
...
+    return true
```
Comment (raymondfeng) [inline lib/mssql.js]: Should we make a callback here?

File: lib/mssql.js [hunk 3]
```diff
@@ -16,12 +16,16 @@ exports.initialize = function initializeSchema(dataSource, callback) {
...
+  } else {
+      driver.connect(function(err, connection) {
+      dataSource.client = connection;
+      dataSource.connector.dataSource = dataSource;
+      dataSource.connector.tableNameID = dataSource.settings.tableNameID;
```
Comment (raymondfeng) [inline lib/mssql.js]: It won't be necessary to execute the 3 lines above if we set up connector.dataSource/tableNameID before hand.

**Comment by jannyHou** [comment]
@slnode test please

File: lib/mssql.js (hunk 1 above)
Comment (jannyHou) [inline lib/mssql.js]: Sounds good. Changed.

**Comment by jannyHou** [comment]
@raymondfeng Please review:) Thanks.

I applied your feedback: https://github.com/strongloop/loopback-connector-mssql/pull/80#discussion_r59791888
Other changes are same as in [loopback-connector-mongodb](https://github.com/strongloop/loopback-connector-mongodb/pull/241)

**Comment by raymondfeng** [comment]
LGTM

**Pull request merged**